### PR TITLE
Routes with test are finished!

### PR DESCRIPTION
**Pull request by sewelol**

### CODE REVIEW

File: .settings/launch.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{
```
Comment (fsauter) [inline .settings/launch.json]: Do we need this file?

File: Gemfile [hunk 2]
```diff
@@ -78,7 +80,7 @@ group :development, :test do
   gem 'web-console', '~> 2.0'
 
   # Spring speeds up development by keeping your application running in the background. Read more: https://github.com/rails/spring
-  gem 'spring'
+  #gem 'spring'
```
Comment (fsauter) [inline Gemfile]: Why commented out spring?

File: Gemfile [hunk 3]
```diff
@@ -14,6 +14,8 @@ gem 'coffee-rails', '~> 4.1.0'
 # See https://github.com/sstephenson/execjs#readme for more supported runtimes
 gem 'therubyracer', platforms: :ruby
 
+# Use jquery as the JavaScript library
+gem 'jquery-rails'
```
Comment (fsauter) [inline Gemfile]: We have this already in line 52

File: app/assets/javascripts/articles.coffee [hunk 4]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
Comment (fsauter) [inline app/assets/javascripts/articles.coffee]: .coffeee files can be removed

File: app/models/trip.rb [hunk 5]
```diff
@@ -41,6 +43,10 @@ def geocode_end_loc
   def self.search(title_search, location_search, tag_search, date_beg, date_end)
     trips = []
 
+    if name_search.present?
```
Comment (fsauter) [inline app/models/trip.rb]: User search was moved into User model, remove 46-48

File: app/views/articles/show.html.erb [hunk 6]
```diff
@@ -21,7 +21,7 @@
             <% end %>
           </div>
           <div class="header-section">
-            <h1><%= @article.title %></h1>
+            <h1><%= @route.title %></h1>
```
Comment (fsauter) [inline app/views/articles/show.html.erb]: Route != article

File: app/views/equipment_items/show.html.erb [hunk 7]
```diff
@@ -0,0 +1,94 @@
+<% content_for :page_title do %>
```
Comment (fsauter) [inline app/views/equipment_items/show.html.erb]: Delete this file.

**Comment by sewelol** [comment]
removed the coffee files and settings.json and added spring gem :)

File: app/views/events/edit.html.erb [hunk 8]
```diff
@@ -1,4 +1,7 @@
 <% page_title_t 'trip_events_edit' %>
```
Comment (fsauter) [inline app/views/events/edit.html.erb]: I'm not sure if this should be changed with this merge request (IANS)

File: app/views/events/index.html.erb [hunk 9]
```diff
@@ -1,4 +1,7 @@
 <% page_title_t 'trip_events' %>
+<% content_for :page_title do %>
```
Comment (fsauter) [inline app/views/events/index.html.erb]: IANS

File: Gemfile.lock [hunk 10]
```diff
@@ -214,7 +214,6 @@ GEM
       json (~> 1.8)
```
Comment (fsauter) [inline Gemfile.lock]: After all run bundle install and commit the lock file.

File: app/assets/javascripts/requests.coffee [hunk 11]
```diff
@@ -1,3 +0,0 @@
-# Place all the behaviors and hooks related to the matching controller here.
```
Comment (fsauter) [inline app/assets/javascripts/requests.coffee]: coffee file :)

File: app/helpers/routes_helper.rb [hunk 12]
```diff
@@ -0,0 +1,2 @@
+module RoutesHelper
```
Comment (fsauter) [inline app/helpers/routes_helper.rb]: we can delete empty helpers.

File: config/locales/en.yml [hunk 13]
```diff
@@ -237,6 +259,11 @@ en:
   dropbox_auth_not_approved_error:  "Not approved? Why not, bro?"
   dropbox_auth_provider_error:      "Strange error."
   dropbox_error:                    "Error communicating with Dropbox servers."
+
+  # friendship
+  friendship_added:                  "Added friend '%{name}'."
```
Comment (fsauter) [inline config/locales/en.yml]: friendship stuff should be not added in this merge request

File: app/controllers/waypoints_controller.rb [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+  end
+
+
+  def create
+    @trip = Trip.find(params[:trip_id])
+
+    @waypoint = Route.new(params[:trip_id, :route_id]).waypoints
+
+    if @route.update_attribute(params[:trip_id, :route_id])
+      return redirect_to root_url :notice => "waypoint added sucessfully"
```
Comment (ritualz) [inline app/controllers/waypoints_controller.rb]: Remove notice

File: config/routes.rb [hunk 15]
```diff
@@ -23,6 +23,14 @@
     end
     resources :events
     resources :articles
+    resources :routes do
+      resources :waypoints
+    end
+    resources :equipment_lists
```
Comment (fsauter) [inline config/routes.rb]: the routes should only update the "routes + waypoints"

File: public/app/css/main.css [hunk 16]
```diff
@@ -1275,10 +1275,14 @@ img.full-bg.full-bg-bottom {
 }
 
 .calendar-events li:hover {
-    cursor: move;
```
Comment (fsauter) [inline public/app/css/main.css]: nope!!

File: app/views/layouts/_trip_sidebar.html.erb [hunk 17]
```diff
@@ -11,8 +11,10 @@
     <a href="<%=trip_participants_path(@trip)%>"><i class="gi gi-parents sidebar-nav-icon"></i><span class="sidebar-nav-mini-hide"><%=t('trip_participants')%></span></a>
   </li>
 
-  <li class="">
-    <a href="#"><i class="gi gi-direction sidebar-nav-icon"></i><span class="sidebar-nav-mini-hide">Routes</span></a>
+  <li class="<%=current_page?(trip_routes_path(@trip)) ? 'active' : ''%>">
+    <%= link_to trip_routes_path(@trip) do %>
+        <i class="gi gi-direction sidebar-nav-icon"></i><span class="sidebar-nav-mini-hide">Routes</span>
```
Comment (ritualz) [inline app/views/layouts/_trip_sidebar.html.erb]: Translation

File: public/assets/application-3dd084d64bfd88b29ac15846d609ad7e.css [hunk 18]
```diff
@@ -0,0 +1,806 @@
+@charset "UTF-8";/* required styles */
```
Comment (fsauter) [inline public/assets/application-3dd084d64bfd88b29ac15846d609ad7e.css]: this is a generated file and should be removed or?

File: spec/views/map/new.html.erb_spec.rb [hunk 19]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
```
Comment (fsauter) [inline spec/views/map/new.html.erb_spec.rb]: remove map spec

File: spec/helpers/friendships_helper_spec.rb [hunk 20]
```diff
@@ -0,0 +1,15 @@
+require 'rails_helper'
```
Comment (fsauter) [inline spec/helpers/friendships_helper_spec.rb]: remove friendship helper

File: spec/helpers/articles_helper_spec.rb [hunk 21]
```diff
@@ -0,0 +1,15 @@
+require 'rails_helper'
```
Comment (fsauter) [inline spec/helpers/articles_helper_spec.rb]: remove this helper

File: spec/features/search_spec.rb [hunk 22]
```diff
@@ -1,79 +0,0 @@
-require 'rails_helper'
```
Comment (fsauter) [inline spec/features/search_spec.rb]: why is this file deleted?

File: spec/controllers/waypoints_controller_spec.rb [hunk 23]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
```
Comment (fsauter) [inline spec/controllers/waypoints_controller_spec.rb]: no tests? we can delete it then :)

File: app/models/trip.rb [hunk 24]
```diff
@@ -9,6 +9,8 @@ class Trip < ActiveRecord::Base
   has_many :events, :dependent => :delete_all
   has_many :articles, :dependent => :delete_all
   has_many :discussions, :dependent => :delete_all
+  has_many :routes, :dependent => :delete_all
+  has_many :articles, :dependent => :delete_all
```
Comment (fsauter) [inline app/models/trip.rb]: remove the article line

File: app/controllers/trips_controller.rb [hunk 25]
```diff
@@ -39,6 +39,10 @@ def create
     end
   end
 
+  def search
+    @user, @trips = Trip.search(params[:title_search], params[:location_search], params[:tag_search], params[:date_beg], params[:date_end], params[:name_search])
```
Comment (fsauter) [inline app/controllers/trips_controller.rb]: Revert this file to current, has nothing to do with routes

File: app/controllers/trip_resource_controller.rb [hunk 26]
```diff
@@ -32,7 +32,7 @@ def set_trip
     @trip = Trip.find params[:trip_id]
```
Comment (fsauter) [inline app/controllers/trip_resource_controller.rb]: revert.

File: Gemfile [hunk 27]
```diff
@@ -52,8 +52,7 @@ gem 'geocoder'
 gem 'jquery-rails'
 gem 'jquery-turbolinks'
 
-
-# Kaminari, pageinate
```
Comment (fsauter) [inline Gemfile]: can be reverted to have less changes :)

File: config/locales/en.yml [hunk 28]
```diff
@@ -8,7 +8,7 @@
 #
 # In views, this is aliased to just `t`:
 #
-#
+#
```
Comment (fsauter) [inline config/locales/en.yml]: can be reverted for a clean commit.

File: spec/helpers/routes_helper_spec.rb [hunk 29]
```diff
@@ -0,0 +1,15 @@
+require 'rails_helper'
```
Comment (ritualz) [inline spec/helpers/routes_helper_spec.rb]: Remove file?
